### PR TITLE
refactor EnvUtil to handle corner cases (#1628)

### LINKED ISSUE #1628: Refactor EnvUtil to handle corner cases
### Component

JKube Kit

### Description
Refactor `extractLargerVersion()` to handle corner cases like If the given versions are invalid, `null` should be returned otherwise return the valid version.
https://github.com/eclipse/jkube/blob/61345823be63c73cf334649e33e9c5146f9d5f7a/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java#L81

### Expected Behavior
`null` should be returned if given versions are invalid.

This is specified in the Javadoc:
https://github.com/eclipse/jkube/blob/4cef8d361a73f04161699dd149383b70f60de6ca/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java#L71-L81

### Acceptance Criteria
- [x] Tests should be evaluated and refactored if applicable (optional)

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description
Fixes #1628 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by anurag-rajawat** [comment]
@manusa and @rohanKanojia kindly review this because it blocked #1621

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 1]
```diff
@@ -139,17 +137,15 @@ public void testGreaterOrEqualsVersionWhenFalse() {
...
+    @Test
     public void testGreaterOrEqualsVersionCornerCase() {
         //Given
         String versionA = "asdw4.0.2";
         String versionB = "3.1.1.0{0.1}";
         //When
         String result = EnvUtil.extractLargerVersion(versionA, versionB);
         //Then
-        fail("Exception should have thrown");
+        assertThat(result).isNull();
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:148]: You should modify the implementation to throw exception when version is invalid

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 2]
```diff
@@ -102,6 +105,37 @@ public void testExtractLargerVersion() {
...
+        String versionA = "3.a.b.c";
+        String versionB = "4.0.2";
+        // When
+        String result = EnvUtil.extractLargerVersion(versionA, versionB);
+        // Then
+        assertThat(result).isEqualTo("4.0.2");
+    }
+
+    @Test
+    public void testGreaterOrEqualsVersionWhenBIsInvalid() {
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:129]:
```suggestion
    public void extractLargerVersion_whenBIsInvalid_shouldReturnA() {
```

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 3]
```diff
@@ -102,6 +105,37 @@ public void testExtractLargerVersion() {
...
+    public void testGreaterOrEqualsVersionWhenBothAreInvalid() {
+        //Given
+        String versionA = "asdw4.0.2";
+        String versionB = "3.1.1.0{0.1}";
+        //When & Then
+        assertThrows(IllegalArgumentException.class, () -> EnvUtil.extractLargerVersion(versionA, versionB));
+    }
+
+    @Test
+    public void testGreaterOrEqualsVersionWhenAIsInvalid() {
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:118]:
```suggestion
    public void extractLargerVersion_whenAIsInvalid_shouldReturnB() {
```

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 4]
```diff
@@ -102,6 +105,37 @@ public void testExtractLargerVersion() {
         assertThat(result).isEqualTo("4.0.0.1");
     }
 
+    @Test
+    public void testGreaterOrEqualsVersionWhenBothAreInvalid() {
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:109]:
```suggestion
    public void extractLargerVersion_whenBothInvalid_thenThrowsException() {
```

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java (hunk 1 above)
Comment (manusa) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java]: The issue changes the spec to:

> ### Expected Behavior
> 
> `null` should be returned if given versions are invalid.

Maybe we can elaborate on the reasoning behind this

Comment (manusa) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java]: I just updated the Issue description to be clearer:
image

**Pull request merged**